### ISSUE TITLE
feat(pagination): add skipping results support

### ISSUE THREAD

**Issue by vallahaye** [einride/aip-go#92]
Hello,

The [AIP-158](https://google.aip.dev/158#skipping-results) specification defines an optional `skip` parameter that allows you to skip results when listing a collection.
At the moment, this parameter is not taken into account by the paging module and I think it would be nice if it was supported.

Thank you for open sourcing your implementation.

Sincerely

**Comment by ericwenn** [comment]
How would you like to see it supported? Currently the paging package performs parsing and creating page tokens, and at a glance `skip` seems unrelated to this.

**Comment by vallahaye** [comment]
Hello again,

Sorry for the delay in responding. As I understand the [specification](https://google.aip.dev/158#skipping-results), the `skip` parameter, when present, should be taken into account and added to the offset when parsing the page token.
This field must also be cleared before calculating the request checksum, because the user can change its value between successive calls to the same collection to "jump" from one page to another without requesting all the intermediate pages.

**Comment by ericwenn** [comment]
Hi,

I'm skeptical about changing the `Offset` of `PageToken` struct based on the value of `skip` because it would mean that if we add another page token implementation which does not directly include an offset (many databases natively expose a cursor/page token), it would lead to two APIs that behave differently.

However, we could (and should) ignore the value of `skip` when calculating the checksum.
That would mean you would use the API like this:
```go
pageToken, err := pagination.ParsePageToken(request)
// ...
result, err := s.Storage.ListBooks(ListBooksQuery{
  Offset: pageToken.Offset + request.GetSkip(),
})
// ...
response.NextPageToken = pageToken.Next(request).String()
```

What do you think?

**Comment by vallahaye** [comment]
You are right, from this point of view it makes perfect sense not to change the parsing function.
Letting the user add the `skip` value to the `offset` himself also makes the code less opaque and easier to understand.
I agree with your solution.

**Comment by ericwenn** [comment]
Sorry for late reply! Here is a PR that ignores `skip` when calculating the checksum:
https://github.com/einride/aip-go/pull/93

**Comment by ericwenn** [comment]
Solved by #93
